### DISCUSSION TITLE
Working directory for job : unable to make it working

### DISCUSSION THREAD

**Discussion by LeRatierBretonnien** [gocd/gocd#12650] [Q&A] [answered]
Hi !
I have a job with 3 sequential tasks : 
1 - Script execution at root level of the checkout (ie "working directory" = empty)
2 - make execution at root level of the checkout (ie "working directory" = empty)
3 - do another make execution in a sub folder of the checkout (ie "working directory" = pointing to a sub-folder)

-> The task 3 has always an error, saying that the folder is not found. But the folder exists in the checkout copy !

**Comment by chadlwilson** [comment]
You’ll need to share more specifics about your job and task definition and the error you see than this. There’s no way to know what you’re doing wrong from the description above.

Check that your material definition in the pipeline has no ‘alternate checkout path’ set, or the checkout path you expect, and make sure that it reflected in your task working directories.

the job console log will show you were materials are being cloned to.

**Comment by sriramnrn** [comment]
On Fri, Mar 29, 2024 at 5:49 PM LeRatierBretonnien ***@***.***>
wrote:

> Hi !
> I have a job with 3 sequential tasks :
> 1 - Script execution at root level of the checkout (ie "working directory"
> = empty)
> 2 - make execution at root level of the checkout (ie "working directory" =
> empty)
> 3 - do another make execution in a sub folder of the checkout (ie "working
> directory" = pointing to a sub-folder)
>
-> The task 3 has always an error, saying that the folder is not found. But
> the folder exists in the checkout copy !
>
What does the command contain?

Assume that the sub-folder is called "tests", and you need to run "make
all-tests", then you would need to specify
working directory: tests
command: make
arguments: all-tests

That is, the command should not contain the sub-folder (ie. it should not
be tests/make)

**Comment by LeRatierBretonnien** [comment]
Thanks for your anwsers.
I have no "'alternate path" : 
![image](https://github.com/gocd/gocd/assets/1085008/48f2ef57-8541-459b-a1e8-e661d0e9825c)

Here are my jobs configuration : 
![image](https://github.com/gocd/gocd/assets/1085008/962d69e5-f743-4e48-b5b8-74b5731816bc)

The first 2 jobs are performed properly (ie at the root of the checkout), but the third one is failing, saying that there is no "Makefile_deps" file. But the file is present in the working folder.

Any hint or idea?

**Reply by chadlwilson** [reply]
What happens if you add a task like this before step 3?
Command =`ls`
Args = `derivated_products/common/widgets/SDK`
Working dir = empty

can you share the actual full job console output of this new task and the failing task?

**Comment by LeRatierBretonnien** [comment]
Good idea, thanks.
Here is the output
![image](https://github.com/gocd/gocd/assets/1085008/1fa95089-6795-4645-85e4-e85e830c986d)

**Reply by chadlwilson** [reply, marked as answer]
In your third task config check the two arguments are each on a *separate line* in the dialog.

Based on that output, it looks like they are not - note how they are printed as “-f Makefile_deps” with quotes in the task, meaning they are being sent as a single arg to make. Also note how the error is `No rule to make target ' Makefile_deps'` with an extra space before it. This is the problem you have.

**Reply by chadlwilson** [reply]
I don’t know why make strips the -f from the arg, that must just be how it works.
